yash-datalake/azureDeploy.json

```diff
@@ -0,0 +1,545 @@
+			}
+		},
+		"adlStoreName": {
+			"type": "string",
+			"defaultValue": "datalakedemoadls1a",
+			"metadata": {
+				"description": "The name of the Data Lake Store account to create."
+			}
+		},
+		"vmSize": {
+			"type": "string",
+			"allowedValues": [
+				"Standard_A1",
+				"Standard_A2",
+				"Standard_A3",
+				"Standard_A4",
+				"Standard_D1_v2",
+				"Standard_D2_v2",
+				"Standard_D3_v2",
+				"Standard_D4_v2"
+			],
+			"defaultValue": "Standard_D1_v2",
+			"metadata": {
+				"description": "Size of vm"
+			}
+		},
+		"vm_username": {
+			"type": "string",
+			"defaultValue": "azuredemo",
+			"metadata": {
+				"description": "Username for the Virtual Machine."
+			}
+		},
+		"vm_password": {
+			"type": "string",
+			"metadata": {
+				"description": "Password for the Virtual Machine."
+			},
+			"defaultValue": "Yash@1234"
+		},
+		"OptionalWizardInstall": {
+			"type": "string",
+			"allowedValues": [
+				"Yes",
+				"No"
+			],
+			"defaultValue": "Yes",
+			"metadata": {
+				"description": "Select whether the VM should be in production or not."
+			}
+		},
+		"dataFactoryName": {
+			"type": "string",
+			"metadata": {
+				"description": "Name of the data factory. Must be globally unique."
+			},
+			"defaultValue": "[toLower(uniqueString(resourceGroup().id, 'datafactorya'))]"
+		},
+		"servicePrincipalId": {
+			"type": "string",
+			"defaultValue": "null",
+			"metadata": {
+				"description": "The ID of the service principal that has permissions to create HDInsight clusters in your subscription."
+			}
+		},
+		"servicePrincipalKey": {
+			"type": "string",
+			"defaultValue": "null",
+			"metadata": {
+				"description": "The access key of the service principal that has permissions to create HDInsight clusters in your subscription."
+			}
+		},
+		"_artifactsLocationSasToken":{
+			"type": "SecureString",
+			"defaultValue": ""
+		}
+
+	},
+	"variables": {
+		"storageAccountName": "[toLower(uniqueString(resourceGroup().id, 'storagea'))]",
+		"blobContainer": "raw",
+		"storageAccountType": "Standard_LRS",
+		"maximumThroughputUnits": 8,
+		"DataCleaningInputStream": "datalakerawstreama",
+		"DataCleaningOutputStream": "datalakecleanstreama",
+		"captureNameFormat": "{Namespace}/{EventHub}/{PartitionId}/{Year}/{Month}/{Day}/{Hour}/{Minute}/{Second}",
+		"DataCleaningStreamAnalyticsJob": "cleanOrdersStreamAnalyticsJoba",
+		"DataAggregatorStreamAnalyticsJob1": "sellingProductPerMinJoba",
+		"DataAggregatorStreamAnalyticsJob2": "geoLocationsJoba",
+		"DataAggregatorStreamAnalyticsJob3": "ageGroupWiseRevenueJoba",
+		"appName": "azuredatalakeuiappa",
+		"dnsLabelPrefix": "[concat('datalake',uniquestring(resourceGroup().id,deployment().name))]",
+		"ES_Function_name": "elkstackfuntiona",
+		"serverfarms_EastUS2Plan_name": "EastUS2Plana",
+		"config_web_name_4": "web",
+		"_artifactsLocation": "https://raw.githubusercontent.com/ajos1993/azure-quickstart-templates/master/yash-datalake/",

```

Comment:
If we use deployment().properties.templateLink.uri() or other method then it is unable to refer our nested templates. I think once its in Azure repo it will work. Can you suggest any other change which will work out in the current situation?

Reply:
Updated according to best practices